xl-objects.go

```diff
@@ -150,6 +156,8 @@ func (xl xlObjects) GetObject(bucket, object string, startOffset int64) (io.Read
 	if !IsValidObjectName(object) {
 		return nil, ObjectNameInvalid{Bucket: bucket, Object: object}
 	}
+	nsMutex.RLock(bucket, object)
+	defer nsMutex.RUnlock(bucket, object)
```

Comment:
It is better to perhaps "RUnlock()" after its known that is not a Multipart object, since there would be multiple `RLock()` on the same bucket, object..